\textbf{Motivation:} Statistical analyses of high-throughput sequencing data have re-shaped the biological sciences. In spite of myriad advances, recovering interpretable biological signal from data corrupted by technical noise remains a prevalent open problem. Several classes of procedures, among them classical dimensionality reduction techniques, and others incorporating subject-matter knowledge, have provided effective advances; however, no procedure currently satisfies the dual objectives of recovering stable and relevant features simultaneously. 

\textbf{Results:} Inspired by recent proposals for making use of control data in the removal of unwanted variation, we propose a variant of principal component analysis that extracts sparse, stable, interpretable, and relevant biological signal. The new methodology is compared to competing
dimensionality reduction approaches through a simulation study as well as via analyses of several publicly available protein expression, microarray gene expression, and single-cell transcriptome sequencing datasets.

\textbf{Availability:} A free and open-source software implementation of the methodology, the \texttt{scPCA} \texttt{R} package, is made available via the \href{https://bioconductor.org/packages/release/bioc/html/scPCA.html}{Bioconductor} Project. Code for all analyses presented in the paper are also made available via \href{https://github.com/PhilBoileau/EHDBDscPCA}{GitHub}.

%\textbf{SUPPLEMENTATY MATERIAL:} Supplementary data are available at Bioinformatics online. (Need to include href) [NH: fine to add this after acceptance I think]
