\documentclass{article}
\usepackage{arxiv}
\input{_preamble}
\externaldocument{main}

\title{\textit{Supplementary Materials for}\\ Exploring High-Dimensional Biological Data with Sparse Contrastive Principal Component Analysis}

\author{
  Philippe Boileau \\
  Graduate Group in Biostatistics,\\
  University of California, Berkeley\\
  \texttt{philippe\_boileau@berkeley.edu} \\
  \And
  Nima S.~Hejazi \\
  Graduate Group in Biostatistics and\\
  Center for Computational Biology,\\
  University of California, Berkeley\\
  \texttt{nhejazi@berkeley.edu} \\
  \And
  Sandrine Dudoit\\
  Department of Statistics, \\
  Division of Biostatistics, and\\
  Center for Computational Biology,\\
  University of California, Berkeley\\
  \texttt{sandrine@stat.berkeley.edu} \\
}

\begin{document}\maketitle
\beginsupplement

\newpage
\section{Algorithm for scPCA}

\begin{algorithm}[!htbp]\label{algo1}
  \dontprintsemicolon
  \linesnotnumbered
  \KwResult{Produces a sparse low-dimensional representation of the target data,
    $\mathbf{X}_{n \times p}$, by contrasting the variation of
    $\mathbf{X}_{n \times p}$ and some background data, $\mathbf{Y}_{m \times p}$,
    while applying an $\ell_1$ penalty to the loadings generated by cPCA.}
  \SetKwInOut{Input}{Input}\SetKwInOut{Output}{Output}
  \Input{
  \begin{itemize}
    \item[] target dataset: $\mathbf{X}$
    \item[] background dataset: $\mathbf{Y}$
    \item[] binary variable indicating whether to column-scale the data: \texttt{scale}
    \item[] vector of possible contrastive parameters: $\gamma = (\gamma_1, \ldots, \gamma_s)$
    \item[] vector of possible $\ell_1$ penalty parameters: $\lambda_1 = (\lambda_{1,1}, \ldots, \lambda_{1, d})$
    \item[] number of sparse contrastive principal components to compute: $k$
    \item[] clustering method: \texttt{cluster\_meth}
    \item[] number of clusters: \texttt{ncluster}
  \end{itemize}
  }
  \BlankLine
  Center (and \texttt{scale} if so desired) the columns of $\mathbf{X}$, $\mathbf{Y}$ \;
  Calculate the empirical covariance matrices:
  $\mathbf{C_X}_{p \times p} \coloneqq \frac{1}{n}\mathbf{X}^\top\mathbf{X}, \;
  \mathbf{C_Y}_{p \times p} \coloneqq \frac{1}{m}\mathbf{Y}^\top\mathbf{Y}$\;
  \For{each $\gamma_i \in \gamma$}{
    \For{each $\lambda_{1,j} \in \lambda_1$}{
      Compute the contrastive covariance matrix $\mathbf{C}_{\gamma_i}=\mathbf{C_X} -
      \gamma_i \mathbf{C_Y}$\;
      Compute the positive-semidefinite approximation of $\mathbf{C}_{\gamma_i}$, $\widetilde{\mathbf{C}}_{\gamma_i}$\;
      Apply SPCA to $\widetilde{\mathbf{C}}_{\gamma_i}$ for $k$ components with $\ell_1$ penalty $\lambda_{1, j}$\;
      Generate a low-dimensional representation by projecting $\mathbf{X}_{n \times p}$ on the sparse loadings of SPCA\;
      Normalize the low-dimensional representation produced to be on the unit hypercube\;
      Cluster the normalized low-dimensional representation using \texttt{cluster\_meth} with \texttt{ncluster}\;
      Compute and record the clustering strength criterion associated with
      $(\gamma_i, \lambda_{1, j})$ \;
    }
  }
  Identify the combination of hyperparameters maximizing the clustering strength
  criterion: $\gamma^{\star},\: \lambda_1^{\star}$ \;
  \Output{The low-dimensional representation of the target data given by
    $(\gamma^{\star}, \lambda_1^{\star})$, an $n \times k$ matrix;
    the $p \times k$ matrix of loadings given by
    $(\gamma^{\star}, \lambda_1^{\star})$;
    contrastive parameter $\gamma^{\star}$;
    $\ell_1$ penalty parameter $\lambda_1^{\star}$
  }
  \caption{scPCA}
\end{algorithm}


\newpage
\section{Algorithm for Cross-Validated scPCA}

\begin{algorithm}[!htbp]\label{algo2}
  \dontprintsemicolon
  %\setalgolined
  \linesnotnumbered
  \KwResult{Produces a sparse low-dimensional representation of the target data,
    $\mathbf{X}_{n \times p}$, by contrasting the variation of
    $\mathbf{X}_{n \times p}$ and some background data, $\mathbf{Y}_{m \times p}$,
    while applying an $\ell_1$ penalty to the loadings generated by cPCA.}
  \SetKwInOut{Input}{Input}\SetKwInOut{Output}{Output}
  \Input{
  \begin{itemize}
    \item[] target dataset: $\mathbf{X}$
    \item[] background dataset: $\mathbf{Y}$
    \item[] binary variable indicating whether to column-scale the data:
      \texttt{scale}
    \item[] vector of possible contrastive parameters:
      $\gamma = (\gamma_1, \ldots, \gamma_s)$
    \item[] vector of possible $\ell_1$ penalty parameters: $\lambda_1 = (\lambda_{1,1}, \ldots, \lambda_{1, d})$
    \item[] number of sparse contrastive principal components to compute: $k$
    \item[] clustering method: \texttt{cluster\_meth}
    \item[] number of clusters: \texttt{ncluster}
    \item[] number of cross-validation folds: $W$
  \end{itemize}
  }
  \BlankLine
   For $\mathbf{X}_{n \times p}$, randomly partition the index set $\{1, \ldots,
  n\}$ into $W$ validation sets, $\mathcal{W}_1^x, \ldots, \mathcal{W}_W^x$, of (approximately) the
  same size (i.e., $\bigcup_{w=1}^W \mathcal{W}_w^x = \{1, \ldots, n\}$;
  $\mathcal{W}_w^x
  \cap \mathcal{W}_{w'}^x = \emptyset, \quad \forall w,w' \in \{1, \ldots, W\}$).
  Denote the corresponding training sets by $\mathcal{T}_w^x = \{1, \ldots, n\}
  \setminus \mathcal{W}_w^x$. For $\mathbf{Y}_{m \times p}$, randomly partition the
  index set $\{1, \ldots, m\}$ into $W$ validation sets, $\mathcal{W}_1^y, \ldots,
  \mathcal{W}_W^y$, of (approximately) the same size (i.e., $\bigcup_{w=1}^W \mathcal{W}_w^y = \{1,
      \ldots, m\}$; $\mathcal{W}_w^y \cap \mathcal{W}_{w'}^y = \emptyset, \quad
      \forall
      w,w' \in \{1, \ldots, W\}$). Denote the corresponding training sets by
  $\mathcal{T}_w^y = \{1, \ldots, m\} \setminus \mathcal{W}_w^y$. Denote by
$\mathbf{X}_{\mathcal{T}_w^x}$ the $\lvert \mathcal{T}_w^x \rvert\
  \times p$ submatrix of $\mathbf{X}$ for training set $\mathcal{T}_w^x$ and by
$\mathbf{Y}_{\mathcal{T}_w^y}$ the $\lvert \mathcal{T}_w^y \rvert
  \times p$ submatrix of $\mathbf{Y}$ for training set $\mathcal{T}^y_w$. Define similarly
  $\mathbf{X}_{\mathcal{W}_w^x}$ and $\mathbf{Y}_{\mathcal{W}_w^y}$ for the validation
  sets. Note that $\mathbf{Y}_{\mathcal{W}_w^y}$ is defined explicitly solely to
  avoid ambiguity; it plays no role in subsequent developments.\; \BlankLine
  \For{each $w$ in $\{1, \ldots, W\}$}{
    Center (and \texttt{scale} if so desired) the columns of
      $\{\mathbf{X}_{\mathcal{T}_w^x}, \mathbf{Y}_{\mathcal{T}_w^y} \}$ and
      $\{\mathbf{X}_{\mathcal{W}_w^x}, \mathbf{Y}_{\mathcal{W}_w^y} \}$\;
  Compute the empirical covariance matrices:
    $\mathbf{C_X}_{p \times p} \coloneqq \frac{1}{\lvert \mathcal{T}_w^x
    \rvert}\mathbf{X_{\mathcal{T}_w^x}}^\top\mathbf{X_{\mathcal{T}_w^x}}, \;
  \mathbf{C_Y}_{p \times p} \coloneqq \frac{1}{\lvert \mathcal{T}_w^y
    \rvert}\mathbf{Y_{\mathcal{T}_w^y}}^\top\mathbf{Y_{\mathcal{T}_w^y}}$\;
  \For{each $\gamma_i \in \gamma$}{
    \For{each $\lambda_{1,j} \in \lambda_1$}{
      Compute the contrastive covariance matrix $\mathbf{C}_{\gamma_i}=\mathbf{C_X} -
      \gamma_i \mathbf{C_Y}$\;
      Compute the positive-semidefinite approximation of $\mathbf{C}_{\gamma_i}$, $\widetilde{\mathbf{C}}_{\gamma_i}$\;
      Apply SPCA to $\widetilde{\mathbf{C}}_{\gamma_i}$ for $k$ components with $\ell_1$ penalty $\lambda_{1, j}$\;
      Generate a low-dimensional representation of the target validation set by
        projecting $\mathbf{X}_{\mathcal{W}_w^x}$ on the sparse loadings of SPCA\;
      Normalize the low-dimensional representation produced to be on the unit hypercube\;
      Cluster the normalized low-dimensional representation using \texttt{cluster\_meth} with \texttt{ncluster}\;
      Compute and record the clustering strength criterion associated with $(\gamma_i, \lambda_{1, j})$ \;
    }
  }
  }
  Identify the combination of hyperparameters maximizing the cross-validated
  mean (across all folds $\{1, \ldots, W\}$) of the clustering strength
  criterion: $\gamma^{\star},\: \lambda_1^{\star}$ \;
  \Output{The low-dimensional representation of the target data given by
    $(\gamma^{\star}, \lambda_1^{\star})$, a $n \times k$ matrix;
    the $p \times k$ matrix of loadings given by $(\gamma^{\star},
    \lambda_1^{\star})$; contrastive parameter $\gamma^{\star}$;
    $\ell_1$ penalty parameter $\lambda_1^{\star}$
  }
  \caption{Cross-validated scPCA}
 \end{algorithm}

\newpage

\section{Intuition for the Contrastive Parameter $\gamma$}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/cpca_example_plot}
    \caption{
    {\em Effect of contrastive parameter for cPCA.}
    cPCA as implemented by \citet{Abid2018} was applied to a simulated dataset
    of $n=400$ observations, split across 4 groups, with $p=30$ variables. The
    first 10 variables are distributed as $N(0, 10)$ for all observations. Variables 11  through 20 are distributed as $N(0, 1)$ for Groups 1 an 2,
    and as $N(3, 1)$
    for Groups 3 and 4. Variables 21 through 30 are distributed as $N(-3, 1)$
    for Groups 1 and 3, and as $N(0, 1)$ for Groups 2 and 4. cPCA also takes
    as input a background dataset of $m=400$ observations, with $p=30$
    variables, where the first 10 variables are distributed as $N(0, 10)$, the
    following 10 as $N(0, 3)$, and the remaining 10 as $N(0, 1)$. The results
    of cPCA are then presented for eight increasing values of the contrastive
    parameter $\gamma$ from among 40 loagrithmically spaced values between 0.1 and 100, selected using the semi-automated technique described by \citet{Abid2018}. For the smaller values of the contrastive parameter, the
    noise contained in the first 10 variables of the target data dominates the
    signal contained in variables 11 through 30. As the contrastive parameter
    increases, the signal in the target data set is unmasked. However, once the
    contrastive parameter value becomes larger than $\approx 20$, the
    distinction between groups becomes increasingly poor; the variation
    contained in the background data begins to dominate the variation contained
    in the target data. A virtually identical dataset is presented in the
    supplementary material of \citet{Abid2018}.}
    \label{contrastive_par}
\end{figure}

\FloatBarrier

\newpage

\section{Simulated scRNA-seq Data}\label{sup_sim}

See Section 3.1 %\ref{sim_scRNA-seq}
for information on the simulation model and dataset. cPCA was applied to the dataset using the non-cross-validated hyperparameter tuning framework. The $\ell_1$ penalty parameter was set to 0, and the vector of possible contrastive parameters consisted of 40 logarithmically spaced values between 0.1 and 1,000. scPCA was applied in the same manner as cPCA, though the vector of potential $\ell_1$ penalty parameters consisted of 20 equidistant values between 0.05 and 1. The \texttt{Rtsne R} package was used to create the t-SNE embedding. Two initializations were generated, that is, with and without an initial application of PCA to the simulated data. The embedding employing the initial PCA step retained the 50 leading principal components. For each initialization, the remaining parameters were set to their defaults, as done by \citet{Becht2019}, e.g., \texttt{perplexity = 30} and \texttt{max\_iter = 1000}. The \texttt{theta} parameter was set to \texttt{0} so that exact t-SNE was performed. The embedding produced using the PCA initialization was qualitatively better than that without. Therefore, it is used in Figure \ref{fig:sim} of this manuscript. The \texttt{umap} \texttt{R} package was used to generate the UMAP embedding. As in the quantitative analysis of UMAP performed by \citet{Becht2019}, \texttt{min\_dist} was set to \texttt{0.02}, \texttt{nearest\_neighbors} was set to \texttt{30}, and the Euclidean distance was used as a metric. The SIMLR was produced with a \texttt{k} of 10, as recommended in \citep{Wang2017}, and setting the number of pre-specified clusters to 2.


\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/sim_de_genes}
    \caption{{\em Simulated scRNA-seq data: Differential expression.}
    The 98 differentially expressed genes in the simulated target dataset are ranked in decreasing order of their absolute level of differential expression between groups. In the \textit{Splatter} framework, genes are differentially expressed between groups by way of a group-specific multiplicative factor. Thus, the level of differential expression of any gene between two groups may be computed as the absolute value of the difference between each group's multiplicative factor. We find that all 20 of the genes with non-zero entries in scPCA's first loading vector, highlighted in green, are among the most differentially expressed.}
    \label{fig:sim_de_genes}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/technical_effect_removal_sim}
    \caption{{\em Simulated scRNA-seq data: Average silhouette width comparison.}
    Methods are displayed in decreasing order of ability to remove unwanted technical variation, as measured by the average silhouette width. scPCA produces the densest biological clusters with the least amount of technical noise. The ZINB-WaVE method, when taking into account the batch effect, has a similar performance to scPCA with respect to the removal of unwanted effects, though the biological clusters it produces have lower average silhouette widths. Though cPCA produces denser biological clusters than ZINB-WaVE, it fails to completely remove the batch effect. The remaining methods are unable to disentangle the biological and technical effects. 
    }
    \label{fig:ave_sil_widths}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/simulated_SIMLR.png}
    \caption{
    {\em Simulated scRNA-seq data: SIMLR.}
     SIMLR's two-dimensional embedding produces dense clusters that nearly perfectly split cells into two biologically meaningful groups. However, upon close inspection -- and contrary to what the average silhouette widths suggest -- the batch effect is not removed from the biological groups. Because the reported average silhouette widths misrepresent the method's ability to remove unwanted variation in the data, its results were not included in Figure \ref{fig:ave_sil_widths}.}
    \label{fig:simulated_SIMLR}
\end{figure}

\FloatBarrier

\section{Dengue Microarray Data}\label{sup_dengue}

See Section 3.2 %\ref{dengue_data}
for information on the data. cPCA and scPCA were fit using the non-cross-validated hyperparameter tuning framework. Both methods considered a vector of 40 logarithmically spaced values between 0.1 and 1,000 as potential contrastive parameters. scPCA also used a vector of 20 equidistant values between 0.05 and 1 as potential $\ell_1$ penalty parameters. As with the simulated data, two t-SNE embeddings were generated: one with an initial PCA step (retaining the first 50 principal components) and one without. Due to the small sample size, the \texttt{perplexity} parameter was set to 8. The remaining hyperparameters were set to their defaults, with the exception of \texttt{theta} which was set to 0. Both embeddings were qualitatively identical, and so only that which does not require the initial dimensionality reduction through PCA is presented in this manuscript. The qualitatively best UMAP embedding was found with the \texttt{n\_neighbors} parameter set to 15 and the \texttt{min\_dist} parameter set to 0.2. These parameter values are inspired from those used by \citet{Becht2019}, though the type of data considered are not identical.

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.5\textwidth]{figures/dengue_tsne}
    \caption{
    {\em Dengue microarray data: t-SNE.}
    Similarly to UMAP, t-SNE almost completely separates the convalescent patients from those with some form of dengue. The two main clusters are further split into distinct sub-clusters, perhaps indicating the presence of a batch effect. 
    }
    \label{fig:dengue_tsne}
\end{figure}

% latex table generated in R 3.6.1 by xtable 1.8-4 package
% Mon Oct 21 14:32:37 2019
\begin{longtable}{| l | p{4cm} | p{8cm} | l |}
  \caption{{\em Dengue microarray data: Genes with non-zero weights in the first scPCA loading vector.}}
  \label{tab:dengue_1} \\
  \hline
    & Gene Symbol & Gene Name & Weight \\ 
  \hline
  \endhead
  1 & PRSS33 & protease, serine, 33 & -0.0059 \\ 
  2 & PDZK1IP1 & PDZK1 interacting protein 1 & -0.0347 \\ 
  3 & SDC1 & syndecan 1 & 0.2507 \\ 
  4 & CAV1 & caveolin 1, caveolae protein, 22kDa & 0.0889 \\ 
  5 & GGH & gamma-glutamyl hydrolase (conjugase, folylpolygammaglutamyl hydrolase) & 0.2318 \\ 
  6 & PI3 & peptidase inhibitor 3, skin-derived & -0.0209 \\ 
  7 & BUB1B & budding uninhibited by benzimidazoles 1 homolog beta (yeast) & 0.1242 \\ 
  8 & ZWINT & ZW10 interactor & 0.3984 \\ 
  9 & TUBB2A & tubulin, beta 2A & -0.0004 \\ 
  10 & PTGS2 & prostaglandin-endoperoxide synthase 2 (prostaglandin G/H synthase and cyclooxygenase) & -0.0627 \\ 
  11 & TTK & TTK protein kinase & 0.0201 \\ 
  12 & ORM1 /// ORM2 & orosomucoid 1 /// orosomucoid 2 & -0.0055 \\ 
  13 & CD38 & CD38 molecule & 0.0399 \\ 
  14 & CHI3L1 & chitinase 3-like 1 (cartilage glycoprotein-39) & -0.0384 \\ 
  15 & HLA-DQB1 & major histocompatibility complex, class II, DQ beta 1 & 0.0720 \\ 
  16 & BUB1 & budding uninhibited by benzimidazoles 1 homolog (yeast) & 0.0853 \\ 
  17 & CDK1 & cyclin-dependent kinase 1 & 0.2650 \\ 
  18 & IGH@ /// IGHA1 /// IGHA2 /// IGHD /// IGHG1 /// IGHG3 /// IGHG4 /// IGHM /// IGHV4-31 /// LOC100290146 /// LOC100290528 & immunoglobulin heavy locus /// immunoglobulin heavy constant alpha 1 /// immunoglobulin heavy constant alpha 2 (A2m marker) /// immunoglobulin heavy constant delta /// immunoglobulin heavy constant gamma 1 (G1m marker) /// immunoglobulin heavy constant gamma 3 (G3m marker) /// immunoglobulin heavy constant gamma 4 (G4m marker) /// immunoglobulin heavy constant mu /// immunoglobulin heavy variable 4-31 /// hypothetical protein LOC100290146 /// similar to pre-B lymphocyte gene 2 & 0.0180 \\ 
  19 & IGH@ /// IGHA1 /// IGHA2 /// IGHD /// IGHG1 /// IGHG3 /// IGHG4 /// IGHM /// IGHV3-23 /// LOC100126583 /// LOC100290146 /// LOC652128 & immunoglobulin heavy locus /// immunoglobulin heavy constant alpha 1 /// immunoglobulin heavy constant alpha 2 (A2m marker) /// immunoglobulin heavy constant delta /// immunoglobulin heavy constant gamma 1 (G1m marker) /// immunoglobulin heavy constant gamma 3 (G3m marker) /// immunoglobulin heavy constant gamma 4 (G4m marker) /// immunoglobulin heavy constant mu /// immunoglobulin heavy variable 3-23 /// hypothetical LOC100126583 /// hypothetical protein LOC100290146 /// similar to Ig heavy chain V-II region ARH-77 precursor & 0.1867 \\ 
  20 & NOV & nephroblastoma overexpressed gene & -0.0619 \\ 
  21 & SELENBP1 & selenium binding protein 1 & -0.1315 \\ 
  22 & IGHA1 /// IGHG1 /// IGHM /// LOC100290293 & immunoglobulin heavy constant alpha 1 /// immunoglobulin heavy constant gamma 1 (G1m marker) /// immunoglobulin heavy constant mu /// similar to hCG2042717 & 0.0162 \\ 
  23 & CEP55 & centrosomal protein 55kDa & 0.2863 \\ 
  24 & PBK & PDZ binding kinase & 0.1358 \\ 
  25 & SHCBP1 & SHC SH2-domain binding protein 1 & 0.2901 \\ 
  26 & MGC29506 & plasma cell-induced ER protein 1 & 0.4012 \\ 
  27 & CNTNAP3 & contactin associated protein-like 3 & -0.0494 \\ 
  28 & JAZF1 & JAZF zinc finger 1 & -0.0441 \\ 
  29 & KIAA1324 & KIAA1324 & -0.0962 \\ 
  30 & CDCA2 & cell division cycle associated 2 & 0.3858 \\ 
  31 & KLHL14 & kelch-like 14 (Drosophila) & 0.0801 \\ 
  32 & CYAT1 & cyclosporin A transporter 1 & 0.1657 \\ 
  33 & HLA-DRB1 /// HLA-DRB3 /// HLA-DRB4 /// HLA-DRB5 /// LOC100294036 & major histocompatibility complex, class II, DR beta 1 /// major histocompatibility complex, class II, DR beta 3 /// major histocompatibility complex, class II, DR beta 4 /// major histocompatibility complex, class II, DR beta 5 /// similar to HLA class II histocompatibility antigen, DRB1-7 beta chain & 0.0422 \\ 
  34 & FLJ10357 & protein SOLO & -0.0966 \\ 
   \hline
\end{longtable}

% latex table generated in R 3.6.1 by xtable 1.8-4 package
% Mon Oct 21 14:52:46 2019
\begin{longtable}{| l | p{4cm} | p{8cm} | l |}
  \caption{{\em Dengue microarray data: Genes with non-zero weights in the second scPCA loading vector.}}  \label{tab:dengue_2} \\
  \hline
 & Gene Symbol & Gene Name & Weight \\ 
  \hline
  \endhead
  1 & PRSS33 & protease, serine, 33 & 0.1822 \\ 
  2 & IFI27 & interferon, alpha-inducible protein 27 & -0.0147 \\ 
  3 & PI3 & peptidase inhibitor 3, skin-derived & 0.1692 \\ 
  4 & SLC2A5 & solute carrier family 2 (facilitated glucose/fructose transporter), member 5 & 0.0701 \\ 
  5 & MYOM2 & myomesin (M-protein) 2, 165kDa & -0.0278 \\ 
  6 & HLA-DRB4 & major histocompatibility complex, class II, DR beta 4 & -0.0620 \\ 
  7 & IGH@ /// IGHA1 /// IGHD /// IGHG1 /// IGHG3 /// IGHG4 /// IGHM /// IGHV3-23 /// IGHV4-31 /// LOC100290146 /// LOC100290528 & immunoglobulin heavy locus /// immunoglobulin heavy constant alpha 1 /// immunoglobulin heavy constant delta /// immunoglobulin heavy constant gamma 1 (G1m marker) /// immunoglobulin heavy constant gamma 3 (G3m marker) /// immunoglobulin heavy constant gamma 4 (G4m marker) /// immunoglobulin heavy constant mu /// immunoglobulin heavy variable 3-23 /// immunoglobulin heavy variable 4-31 /// hypothetical protein LOC100290146 /// similar to pre-B lymphocyte gene 2 & 0.4987 \\ 
  8 & IGKV3-20 & Immunoglobulin kappa variable 3-20 & 0.1623 \\ 
  9 & RSAD2 & radical S-adenosyl methionine domain containing 2 & -0.2294 \\ 
  10 & USP18 & ubiquitin specific peptidase 18 & -0.4599 \\ 
  11 & SIGLEC1 & sialic acid binding Ig-like lectin 1, sialoadhesin & -0.2750 \\ 
  12 & KCTD14 & potassium channel tetramerisation domain containing 14 & -0.3142 \\ 
  13 & FAM118A & family with sequence similarity 118, member A & -0.1382 \\ 
  14 &  &  & 0.0269 \\ 
  15 & SLC16A14 & solute carrier family 16, member 14 (monocarboxylic acid transporter 14) & 0.3232 \\ 
  16 & ANKRD22 & ankyrin repeat domain 22 & -0.2800 \\ 
  17 & KLC3 & kinesin light chain 3 & 0.1021 \\ 
  18 & SIGLEC1 & sialic acid binding Ig-like lectin 1, sialoadhesin & -0.0434 \\ 
   \hline
\end{longtable}


\begin{table}
\caption{{\em Dengue microarray data: Gene set enrichment analysis.} The Broad Institute's online gene set enrichment analysis (GSEA) tool was used to identify the ten most significant gene sets based on GO biological processes \citep{Subramanian15545,Liberzon2011,Liberzon2015}.}
\small
\begin{tabular}{|p{5cm} | p{6cm} | p{1.5cm} | p{1.5cm} | p{1.5cm}|}
  \hline
   Gene Set Name & Description & Genes in Overlap & $p$-value & FDR $q$-value \\ 
  \hline
  GO\_DEFENSE\_RESPONSE & Reactions, triggered in response to the presence of a foreign body or the occurrence of an injury, which result in restriction of damage to the organism attacked or prevention/recovery from the infection caused by the attack. & 13 & 1.12 e-8 & 8.2 e-5 \\
  GO\_RESPONSE\_TO\_CYTOKINE & Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a cytokine stimulus. & 10 & 3.15 e-7 & 1.16 e-3 \\
  GO\_MITOTIC\_CELL\_CYCLE\_CHECKPOINT & A cell cycle checkpoint that ensures accurate chromosome replication and segregation by preventing progression through a mitotic cell cycle until conditions are suitable for the cell to proceed to the next stage. & 5 & 8.39 e-7 & 2.06 e-3 \\
  GO\_CYTOKINE\_MEDIATED\_SIGNALING\_PATHWAY & A series of molecular signals initiated by the binding of a cytokine to a receptor on the surface of a cell, and ending with regulation of a downstream cellular process, e.g. transcription. & 8 & 1.38 e-6 & 2.4 e-3 \\
  GO\_PROTEIN\_LOCALIZATION\_TO\_CHROMOSOME\_ME\_CENTROMERIC\_REGION & Any process in which a protein is transported to, or maintained at, the centromeric region of a chromosome. & 3 & 1.63 e-6 & 2.4 e-3 \\
  GO\_CELL\_CYCLE\_CHECKPOINT & A cell cycle process that controls cell cycle progression by monitoring the integrity of specific cell cycle events. A cell cycle checkpoint begins with detection of deficiencies or defects and ends with signal transduction. & 5 & 3.15 e-6 & 3.86 e-3 \\
  GO\_POSITIVE\_REGULATION\_OF\_VASOCONSTRICRICTION & Any process that activates or increases the frequency, rate or extent of vasoconstriction. & 3 & 4.74 e-6 & 4.97 e-3 \\
  GO\_NEGATIVE\_REGULATION\_OF\_METAPHASE\_AN\newline\_ANAPHASE\_TRANSITION\newline\_OF\_CELL\_CYCLE & Any process that stops, prevents or reduces the frequency, rate or extent of metaphase/anaphase transition of cell cycle. & 3 & 8.15 e-6 & 6.58 e-3 \\
  GO\_MITOTIC\_CELL\_CYCLE & Progression through the phases of the mitotic cell cycle, the most common eukaryotic cell cycle, which canonically comprises four successive phases called G1, S, G2, and M and includes replication of the genome and the subsequent segregation of chromosomes into daughter cells. In some variant cell cycles nuclear replication or nuclear division may not be followed by cell division, or G1 and G2 phases may be absent. & 8 & 8.6 e-6 & 6.58 e-3 \\
  GO\_INFLAMMATORY\_RESPONSE & The immediate defensive reaction (by vertebrate tissue) to infection or injury caused by chemical or physical agents. The process is characterized by local vasodilation, extravasation of plasma into intercellular spaces and accumulation of white blood cells and macrophages. & 7 & 9.3 e-6 & 6.58 e-3 \\
 \hline
\end{tabular}
\label{tab:gsea_dengue}
\end{table}


\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/dengue_cpca_centers}
    \caption{
    {\em Dengue microarray data: cPCA.}
    When varying the \textit{a priori} specified number of clusters for cPCA, all four embeddings are virtually identical, suggesting that cPCA is robust to misspecifications of the number of clusters and that optimal contrastive parameters were selected in each case.}
    \label{fig:dengue_cpca_centers}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/dengue_scpca_centers}
    \caption{
    {\em Dengue microarray data: scPCA.}
    When varying the \textit{a priori} specified number of clusters for scPCA, we find that the two-dimensional embeddings are sensitive to this choice. When scPCA is performed on this data with four and five clusters, the results resemble those produced by PCA.}
    \label{fig:dengue_scpca_centers}
\end{figure}

\FloatBarrier

\newpage

\section{Leukemia Patient scRNA-seq Data}\label{sup_aml}

See Section 3.3 %\ref{leukemia_data}
for information on the data. cPCA and scPCA were fit to both patients' data using the non-cross-validated hyperparameter tuning framework. Both methods considered a vector of 40 logarithmically spaced values between 0.1 and 1,000 as potential contrastive parameters. scPCA considered 20 logarithmically spaced values between $1e-9$ and $1$ as potential $\ell_1$ penalty parameters. Two t-SNE embeddings were produced per patient: one with an initial dimension reduction step performed with PCA (retaining the 50 leading principal components) and one without. The remaining parameters for each embedding were set to their defaults, except for \texttt{theta} which was set to 0. UMAP was performed with its default parameters, except for \texttt{n\_neighbors} and \texttt{min\_dist} which were set to 30 and 0.02, respectively. These values match those used by \citet{Becht2019}. The SIMLR embeddings were produced with \texttt{k = 30}, as recommended in \citet{Wang2017} for datasets of this size. The number of pre-specified clusters was set to 2.

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/aml035_SIMLR.png}
    \caption{
    {\em AML Patient 035 scRNA-seq data: SIMLR.} SIMLR fails to produce an informative two-dimensional embedding of the patient's BMMCs. Although a number of dense clusters are formed in the center of the figure, the spread of observations around these clusters renders the visualization uninterpretable.}
    \label{fig:SIMLR_aml_035}
\end{figure}

\begin{table}
\caption{{\em AML Patient 035 scRNA-seq data: Gene set enrichment analysis.}
The Broad Institute's online gene set enrichment analysis tool was used to identify the ten most significant gene sets based on GO biological processes \citep{Subramanian15545, Liberzon2011, Liberzon2015}.}
\small
\begin{tabular}{|p{5cm} | p{6cm} | p{1.5cm} | p{1.5cm} | p{1.5cm}|}
  \hline
   Gene Set Name & Description & Genes in Overlap & $p$-value & FDR $q$-value \\ 
  \hline
  GO\_ESTABLISHMENT\_OF\_PROTEIN\_LOCALIZATIATION\_TO\_ENDOPLASMIC\_RETICULUM & The directed movement of a protein to a specific location in the endoplasmic reticulum. & 33 & 9.82 e-57 & 7.22 e-53 \\ 
  GO\_COTRANSLATIONAL\_PROTEIN\_TARGETING\_TG\_TO\_MEMBRANE & The targeting of proteins to a membrane that occurs during translation. The transport of most secretory proteins, particularly those with more than 100 amino acids, into the endoplasmic reticulum lumen occurs in this manner, as does the import of some proteins into mitochondria. & 32 & 2.09 e-56 & 7.66 e-53 \\ 
  GO\_TRANSLATIONAL\_INITIATION & The process preceding formation of the peptide bond between the first two amino acids of a protein. This includes the formation of a complex of the ribosome, mRNA or circRNA, and an initiation complex that contains the first aminoacyl-tRNA. & 36 & 1 e-54 & 2.46 e-51 \\
  GO\_NUCLEAR\_TRANSCRIBED\_MRNA- \_CATABOLIC\_IC\_PROCESS\_NONSENSE\_MEDIATED\_DECAY & The nonsense-mediated decay pathway for nuclear-transcribed mRNAs degrades mRNAs in which an amino-acid codon has changed to a nonsense codon; this prevents the translation of such mRNAs into truncated, and potentially harmful, proteins. & 32 & 4.32 e-54 & 7.94 e-51 \\
  GO\_PROTEIN\_LOCALIZATION\_TO\_ENDOPLASMICMIC\_RETICULUM & A process in which a protein is transported to, or maintained in, a location within the endoplasmic reticulum. & 33 & 1.45 e-53 & 2.13 e-50 \\
  GO\_VIRAL\_GENE\_EXPRESSION & A process by which a viral gene is converted into a mature gene product or products (proteins or RNA). This includes viral transcription, processing to produce a mature RNA product, and viral translation. & 34 & 7.2 e-51 & 8.82 e-48 \\
  GO\_PROTEIN\_TARGETING\_TO\_MEMBRANE & The process of directing proteins towards a membrane, usually using signals contained within the protein. & 34 & 2.71 e-50 & 2.85 e-47 \\
  GO\_NUCLEAR\_TRANSCRIBED\_MRNA\_CATABOLIC\newline\_IC\_PROCESS & The chemical reactions and pathways resulting in the breakdown of nuclear-transcribed mRNAs in eukaryotic cells. & 33 & 1.22 e-47 & 1.12 e-44 \\
  GO\_ESTABLISHMENT\_OF\_PROTEIN\_LOCALIZATIATION\_TO\_MEMBRANE & The directed movement of a protein to a specific location in a membrane. & 36 & 5.88 e-46 & 4.8 e-43 \\
  GO\_PROTEIN\_TARGETING & The process of targeting specific proteins to particular regions of the cell, typically membrane-bounded subcellular organelles. Usually requires an organelle specific protein sequence motif. & 37 & 2.61 e-43 & 1.92 e-40 \\
 \hline
\end{tabular}
\label{tab:gsea_aml035}
\end{table}


\begin{figure}[!htbp]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/aml027_results}
  \caption{
  {\em AML Patient 027 scRNA-seq data.} Two-dimensional embeddings of the patient's BMMCs produced by PCA, ZINB-WaVE, t-SNE, UMAP, cPCA, and scPCA. cPCA and scPCA produce two-dimensional representations that distinguish between the pre- and post-transplant cells of Patient 027. Although cPCA's embedding contains denser clusters, scPCA's clusters are more distinct -- though they are oddly shaped. This is the result of sparsity: the scPCA embedding is produced with the count data of only five genes.}
  \label{fig:comp_leuk_pat2}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/aml027_SIMLR.png}
    \caption{
    {\em AML Patient 027 scRNA-seq data: SIMLR.} As with AML Patient 035's data, SIMLR produces an uninterpretable representation of the patient's BMMCs.}
    \label{fig:SIMLR_aml_027}
\end{figure}

\FloatBarrier

\section{Mouse Protein Expression Data}\label{sup_mice}

Down Syndrome, the leading genetic cause of intellectual disability \citep{Irving2008}, is the result of trisomy of all or part of the long arm of chromosome 21 \citep{Ahmed2015}. Recently, researchers have begun exploring the use of pharmacotherapies to mitigate these cognitive deficits using the Ts65Dn mouse model \citep{Ahmed2015,Higuera2015}. Though not a perfect model for the study of Down Syndrome, the Ts65Dn displays many relevant neurological phenotypic features, such as deficits in learning and memory \citep{Rueda2012}.

\citet{Ahmed2015} analyzed protein expression in the hippocampus and cortex of Ts65Dn and control mice after exposure to context fear conditioning and Memantine treatment. Memantine, a drug often prescribed to Alzheimer's patients, has been demonstrated to improve performance of the Ts65Dn in tasks that reflect cognitive abilities \citep{Ahmed2015}. The corresponding dataset was made available by Higuera et al. \citep{Higuera2015}. The data consist of normalized expression measures for 77 proteins from subcellular fractions of the cortex assayed from 38 control and 34 Ts65Dn mice. Each protein expression measurement was repeated 15 times (i.e., 15 technical replicates per mouse for each of the 77 proteins), though a small number of replicates contain missing protein expression measurements due to technical artifacts \citep{Higuera2015}. More details on the experimental design are provided in Figure \ref{fig:exp_design}.

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/mice_exp_design}
    \caption{
    {\em Mouse protein expression data: Experimental design.} 
    The control dataset is comprised of protein expression measurements for 15 technical replicates from each of 9 control mice subject to context fear conditioning and given a placebo (red leaf). The target dataset consists of protein expression measurements for 15 technical replicates from each of 9 control mice not subject to context fear condition and given a placebo (purple leaf) and 15 technical replicates from each of 9 trisomic mice not subject to context fear condition and given a placebo (green leaf).}
    \label{fig:exp_design}
\end{figure}

To demonstrate scPCA's capacity to capture biologically meaningful and interpretable variation in protein expression data, the technical replicates of the subset comprising 9 control and 9 Ts65Dn mice not subject to context fear conditioning and given a placebo were designated as the target dataset. The technical replicates of the subset of 9 control mice that were subject to context fear conditioning and given a placebo made up the background dataset, as the variation in their protein expression measurements are believed to be similar to that found in the control mice of the target dataset. The data are identical to those used by \citet{Abid2018} to demonstrate cPCA. PCA, t-SNE, UMAP, cPCA, and scPCA were applied to the target dataset (Figure~\ref{fig:comp_mice}A) to identify differences in protein expression between the control and trisomic mice not exposed to the context fear conditioning experiment. In addition to the target dataset, cPCA and scPCA took as input the column-centered background dataset and specified two clusters \textit{a priori}. cPCA and scPCA were fit using the non-cross-validated hyperparameter tuning framework. Both methods considered a vector of 40 equally spaced values between 0.1 and 1,000 as potential contrastive parameters. scPCA used a vector of 20 equidistant values between 0.05 and 1 as potential $\ell_1$ penalty parameters. t-SNE embeddings were produced with and without an initial dimensionality reduction step step. The remaining variables were left at their defaults, with the exception of \texttt{theta} which was set to 0. The embedding produced without an initial application of PCA produced the best embedding, and so it is presented in the manuscript and supplement. The UMAP embedding was generated with \texttt{n\_neighbors} set to 30 and with \texttt{min\_dist} set to 0.02. The remaining parameters were set to their defaults.

PCA proved incapable of distinguishing between the biological groups of interest. UMAP, cPCA, and scPCA successfully split the control and trisomic mice into virtually distinct clusters, though the number of clusters found by UMAP and cPCA in two dimensions did not match, even when varying the \textit{a priori} specified number of clusters in cPCA (Figure~\ref{fig:mice_cpca_centers}). Comparing the results of UMAP and scPCA, we find that they produce the same number of clusters, but their representations of the global structure are markedly different, even when varying the number of clusters specified \textit{a priori} in scPCA (Figure~\ref{fig:mice_scpca_centers}). The presence of distinct Ts65Dn clusters in UMAP's representation may correspond to technical noise that is diminished in cPCA's and scPCA's embeddings, or may arise from UMAP's inability to dependably capture global structure. We also remark that cPCA and scPCA produce very similar embeddings, up to a rotation; however, the first and second columns of scPCA's loading matrix contain merely 12 and 16 non-zero entries, respectively (Figure~\ref{fig:comp_mice}B). Also note that the the separation of control and trisomic mice by scPCA only occurs in scPC2: the proteins with non-zero weights in the corresponding loading vector include AKT, APP, SOD1, and GSK3, each of which has been associated with Down Syndrome in human or mouse models \citep{Troca-Marn9445,Niceta2015,Gulesserian2001,Isacson2002}.  The full list of proteins with non-zero weights in the first two loading vectors of scPCA are provided in Table \ref{tab:mouse_prot_1} and  Table \ref{tab:mouse_prot_2}.

\begin{figure}[!htbp]
  \centering
  \includegraphics[width=0.9\textwidth]{figures/mice_results}
  \caption{{\em Mouse protein expression data.} \textbf{A} All methods but PCA were capable of separating the control from the trisomic mice, though it is unclear why UMAP splits the Ts65Dn mice into two distinct groups. scPCA's low-dimensional representation of the protein expression data is markedly similar to that of cPCA, up to a rotation, despite relying on only a fraction of non-zero values in the loading matrix. On average, scPCA also produces the tightest clusters. Note: a small group of control mice were clustered with the trisomic mice in the UMAP, cPCA, and scPCA representation, potentially comprising a group of mislabeled mice. \textbf{B} scPCA's leading vectors of loadings are much sparser than those of cPCA, increasing the interpretability of findings and clarity of the visualization. The differing rotations of cPCA and scPCA, in addition to the drastically different weighting scheme of the proteins in their respective loading matrices, may indicate that the contrastive step performed by cPCA does not sufficiently dampen spurious sources of variation in the data.}
  \label{fig:comp_mice}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.5\textwidth]{figures/mice_tsne}
    \caption{
    \textit{Mouse protein expression data: t-SNE.}
    t-SNE produces almost linearly-separable clusters, though these clusters contain many fractured, spurious sub-clusters that do not relate to biological signal.
    }
    \label{fig:mice_tsne}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/mice_cpca_centers}
    \caption{
    \textit{Mouse protein expression data: cPCA.}
    When varying the \textit{a priori} specified number of clusters for cPCA, we find that the two-dimensional embedding is once again robust to misspecifications.
    }
    \label{fig:mice_cpca_centers}
\end{figure}

\begin{figure}[!htbp]
    \centering
    \includegraphics[width=0.9\textwidth]{figures/mice_scpca_centers}
    \caption{
    \textit{Mouse protein expression data: scPCA.}
    Unlike with the dengue microarray data, when varying the \textit{a priori} specified number of clusters for scPCA, we find that the two-dimensional embedding is robust to misspecifications. This may indicate that the sensitivity of the method to this tuning parameter is data-dependent.
    }
    \label{fig:mice_scpca_centers}
\end{figure}

\begin{longtable}{| l | p{4cm} | l |}
  \caption{{\em Mouse protein expression data: Proteins with non-zero weights in the first scPCA loading vector.}} \label{tab:mouse_prot_1} \\
  \hline
   & Protein Symbol & Weight \\ 
 \hline
  1 & ELK & 0.0618 \\ 
  2 & BRAF & -0.1001 \\ 
  3 & RSK & -0.0927 \\ 
  4 & SOD1 & 0.1800 \\ 
  5 & S6 & 0.1281 \\ 
  6 & AcetylH3K9 & 0.3992 \\ 
  7 & RRP1 & 0.0606 \\ 
  8 & Tau & 0.7320 \\ 
  9 & CASP9 & -0.0795 \\ 
  10 & PSD95 & -0.0329 \\ 
  11 & Ubiquitin & -0.3674 \\ 
  12 & H3AcK18 & 0.2958 \\ 
 \hline
\end{longtable}

\newpage
\begin{longtable}{| l | p{4cm} | l |}
  \caption{{\em Mouse protien expression data: Proteins with non-zero weights in the second scPCA loading vector.}}
  \label{tab:mouse_prot_2} \\
  \hline
 & Protein Symbol & Weight \\ 
 \hline
  1 & ELK & -0.2236 \\ 
  2 & AKT & -0.0999 \\ 
  3 & APP & 0.3525 \\ 
  4 & SOD1 & -0.2323 \\ 
  5 & NUMB & 0.4690 \\ 
  6 & P70S6 & 0.1554 \\ 
  7 & GSK3B & 0.3574 \\ 
  8 & PKCG & 0.3978 \\ 
  9 & S6 & 0.3674 \\ 
  10 & RRP1 & 0.0272 \\ 
  11 & GluR4 & 0.0404 \\ 
  12 & IL1B & -0.2664 \\ 
  13 & P3525 & 0.0196 \\ 
  14 & PSD95 & -0.1289 \\ 
  15 & SNCA & -0.0783 \\ 
  16 & H3AcK18 & 0.0169 \\ 
 \hline
 \end{longtable}
 
\newpage

\section{Cross-Validated cPCA and scPCA}\label{cv_algo_example}

The cross-validated (CV) versions of cPCA and scPCA (as implemented in Algorithm~\ref{algo2}) were applied to the datasets presented in the main paper and in the supplement. The hyperparameter grids used by each method are the same as those employed for their non-cross-validated counterparts. See Sections~\ref{sup_sim},~\ref{sup_dengue},~\ref{sup_aml}, and \ref{sup_mice} for details. Five-fold CV was applied to the simulated scRNA-seq data, mouse protein expression data, and AML patient scRNA-seq data (Figures~\ref{fig:cv_sim},~\ref{fig:cv_mice},~\ref{fig:cv_aml035}, and ~\ref{fig:cv_aml027}), and 3-fold CV was applied to the dengue microarray gene expression data (Figure \ref{fig:cv_dengue}). A reduced number of folds was used on the latter dataset since it possessed much fewer observations than the others. For each dataset, the CV-cPCA and CV-scPCA embeddings closely resemble -- and are in some cases identical to -- their non-cross-validated counterparts.

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{figures/sim_cv_results.png}
    \caption{\em{CV-cPCA and CV-scPCA on simulated scRNA-seq data.}}
    \label{fig:cv_sim}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{figures/mice_cv_results.png}
    \caption{\em{CV-cPCA and CV-scPCA on mouse protein expression data.}}
    \label{fig:cv_mice}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{figures/dengue_cv_results.png}
    \caption{\em{CV-cPCA and CV-scPCA on dengue microarray data.}}
    \label{fig:cv_dengue}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{figures/aml035_cv_results.png}
    \caption{\em{CV-cPCA and CV-scPCA on AML Patient 035 scRNA-seq data.}}
    \label{fig:cv_aml035}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = \textwidth]{figures/aml027_cv_results.png}
    \caption{\em{CV-cPCA and CV-scPCA on AML Patient 027 scRNA-seq data}}
    \label{fig:cv_aml027}
\end{figure}
 
 
\section{Running Time Comparison}\label{run_time}

The running times of PCA, t-SNE, UMAP, cPCA, scPCA, ZINB-WaVE, and SIMLR were recorded on the simulated scRNA-seq data from Section \ref{sim_scRNA-seq} and on the scRNA-seq data of AML Patient 035 from Section \ref{leukemia_data}.
If a method's software implementation offered the option to parallelize, four cores were used. The hyperparameters used by the methods on each dataset are identical to those described in their respective sections of the supplement, that is Sections~\ref{sup_sim} and \ref{sup_aml}. The \texttt{microbenchmark R} package was used to track the methods' running times. The comparison is presented in Figures~\ref{fig:rt_sim_sc_rna-seq} and \ref{fig:rt_aml035}.

We note that the iterative algorithm presented in Section 2 is computationally inefficient. However, the scPCA framework is not dependent on \citet{Zou2006}'s optimization procedure; other solutions to the SPCA criterion (Equation \eqref{eq:spca}) can be employed to sparsify the loadings of contrastive covariance matrices. Indeed, more efficient algorithms exist.

In particular, recent work by \citet{erichson2018sparse} provides a scalable algorithm by reformulating the SPCA criterion (Equation (3)) as a value function optimization problem. Instead of requiring an iterative routine to update $\mathbf{B}$ of the alternating algorithm presented in Section~\ref{method}, a single operator is used. This procedure can be sped up further through the use of randomized linear algebra methods to compute $\mathbf{A}$.

These sparsification methods were recently included in the \texttt{scPCA R} package, and the distribution of their running times are included in Figures~\ref{fig:rt_sim_sc_rna-seq} and \ref{fig:rt_aml035}. Using these recently developed methods to solve the SPCA step, the scPCA algorithm's running time is decreased by over an order of magnitude on both datasets. Its running time is similar to that of competing methods when using four cores. Given that the hyperparameter tuning framework is embarrassingly parallel, one can expect even faster computation times when more cores are employed.

\begin{figure}
    \centering
    \includegraphics[width = 0.9\textwidth]{figures/rt_analsysis_sim_data.png}
    \caption{
    {\em Running time comparison: Simulated scRNA-seq data.} Each method was applied to the data five times. Note that \textit{scPCA (var. proj.)} and \textit{scPCA (rand. var. proj.)} correspond to the scPCA algorithms relying on the SPCA procedures detailed in \citet{erichson2018sparse}'s recent work, the latter being being the procedure which relies on randomized techniques. \textit{scPCA (iterative)} pertains to the scPCA method that uses the SPCA algorithm detailed in \citet{Zou2006}. The median running time of each method is reported. The most general of the dimensionality reduction methods, PCA, t-SNE, and UMAP, were at least an order of magnitude faster than all other methods. The remaining method's running times were similar, with the exception of the much slower \textit{scPCA (iterative)}.
    }
    \label{fig:rt_sim_sc_rna-seq}
\end{figure}

\begin{figure}
    \centering
    \includegraphics[width = 0.9\textwidth]{figures/rt_analsysis_aml035.png}
    \caption{
    {\em Running time comparison: AML Patient 035 scRNA-seq data.} Each method was applied to the data three times. The median running time of each method is reported. Compared to the smaller dataset, the contrastive methods presented in the manuscript are competitive with the more general dimensionality reduction methods. Indeed, cPCA's running time is similar to that of UMAP, and the scPCA algorithm relying on random numerical methods for sparsification is faster than t-SNE.
    }
    \label{fig:rt_aml035}
\end{figure}

\newpage
\bibliographystyle{natbib}
\bibliography{appendix_ref}

 \end{document}